src/main/java/de/redsix/pdfcompare/CompareResultImpl.java

```diff
@@ -131,10 +132,14 @@ public synchronized void addPage(final PageDiffCalculator diffCalculator, final
             isEqual = false;
             diffAreas.add(diffCalculator.getDiffArea());
             diffImages.put(pageIndex, diffImage);
+            diffPercentages.put(pageIndex, diffCalculator.getDifferenceInPercent());
             pages++;
-        } else if (environment.addEqualPagesToResult()) {
-            diffImages.put(pageIndex, diffImage);
-            pages++;
+        } else {
+            diffPercentages.put(pageIndex, 0.0);
```

Comment:
This is wrong in the case of found differences but below the allowed percentage.
Should always put the value from `getDifferenceInPercent`. Will fix tomorrow.